### PR TITLE
Remove Plugins:

### PR DESCRIPTION
**Pull request by jdrossl**
- Add new APIs for removing plugins
- Refactor for install plugin to generate a single commit
- Improvements for configuration wiring
- Improvements for dependency scan

https://github.com/craftercms/craftercms/issues/5006

### CODE REVIEW

**Comment by jdrossl** [comment]
Turned to draft because I found an issue that will need some discussion

**Comment by jdrossl** [comment]
XSLT issues will be covered  in https://github.com/craftercms/craftercms/issues/5038

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -4541,6 +4541,97 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /marketplace/remove:
+    post:
+      tags:
+        - marketplace
+      summary: Remove a plugin from the given site
+      description: 'Required permission "install_plugins"'
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Should we have a `remove_plugins` permission?

File: src/main/api/2/studio-api-2.yaml [hunk 2]
```diff
@@ -4541,6 +4541,97 @@ paths:
...
+                    type: array
+                    items:
+                      type: string
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '409':
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:4634]: When would this error happen?

File: src/main/api/2/studio-api-2.yaml [hunk 3]
```diff
@@ -4541,6 +4541,97 @@ paths:
...
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '409':
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:4587]: Don't we need a 404?

File: src/main/api/2/studio-api-2.yaml [hunk 4]
```diff
@@ -4541,6 +4541,97 @@ paths:
...
+                    $ref: '#/components/schemas/ApiResponse'
+                  items:
+                    type: array
+                    items:
+                      type: string
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:4630]: Don't we need a 404 (for site not found)?

File: src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java [hunk 5]
```diff
@@ -93,4 +93,15 @@ public void copyPlugin(String siteId, @ValidateSecurePathParam(name = "path") St
         marketplaceServiceInternal.copyPlugin(siteId, path);
     }
 
+    @Override
+    @HasPermission(type = DefaultPermission.class, action = PERMISSION_INSTALL_PLUGINS)
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java]: Should this be updated to `removePlugin` permission?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml]: We can add it, but I can't think of a use case in which an user should be able to only install or only remove plugins

File: src/main/api/2/studio-api-2.yaml (hunk 2 above)
Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml:4634]: Those are standard errors related to authentication & authorization

File: src/main/api/2/studio-api-2.yaml (hunk 3 above)
Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml:4587]: I will add it

File: src/main/api/2/studio-api-2.yaml (hunk 4 above)
Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml:4630]: I will add it

File: src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java (hunk 5 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java]: same as https://github.com/craftercms/studio/pull/2770#discussion_r728460885

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (sumerjabri) [reply src/main/api/2/studio-api-2.yaml]: I can think of the ability to install but not uninstall as a valid configuration. That gives some users the ability to add but not to remove/break this system. Please add it.

File: src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java (hunk 5 above)
Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java]: Let's add it. See my note earlier.

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml]: which roles should have the permission by default?

Comment (sumerjabri) [reply src/main/api/2/studio-api-2.yaml]: Admin and developer should be defaulted to add/remove.

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**